pkgs/development/python-modules/datalad-next/default.nix

```
@@ -0,0 +1,106 @@
+{
+  annexremote,
+  buildPythonPackage,
+  datalad,
+  datasalad,
+  fetchFromGitHub,
+  git,
+  git-annex,
+  humanize,
+  lib,
+  more-itertools,
+  psutil,
+  pythonOlder,
+  setuptools,
+  openssh,
+  unzip,
+  webdavclient3,
+}:
+
+buildPythonPackage rec {
+  pname = "datalad-next";
+  version = "1.5.0";
+  pyproject = true;
+  disabled = pythonOlder "3.12";
+
+  src = fetchFromGitHub {
+    owner = "datalad";
+    repo = "datalad-next";
+    rev = "refs/tags/${version}";
+    hash = "sha256-fqP6nG2ncDRg48kvlsmPjNBOzfQp9+7wTcGvsYVrRzA=";
+  };
+
+  nativeBuildInputs = [ git ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    annexremote
+    datasalad
+    datalad
+    humanize
+    more-itertools
+  ];
+
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  nativeCheckInputs = [
```

Your tests don't run at all during the build.
You need to add `pytestCheckHook` to `nativeCheckInputs` (in other packages also):

```suggestion
  nativeCheckInputs = [
    ...
    pytestCheckHook
    ...
  ];
```

After that, restart your build.

---

No. That changed when the new PEP builder came. Please see https://nixos.org/manual/nixpkgs/unstable/#buildpythonpackage-function :
> In the [installCheck](https://nixos.org/manual/nixpkgs/unstable/#ssec-installCheck-phase) phase, ${python.interpreter} -m pytest is run.
>By default tests are run because [doCheck = true](https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-doCheck).